daemon/containerd/image.go

```diff
@@ -158,15 +163,17 @@ func (i *ImageService) GetImageManifest(ctx context.Context, refOrID string, opt
 	}
 
 	if containerdimages.IsManifestType(desc.MediaType) {
+		if options.Platform != nil && desc.Platform == nil {
+			return nil, errdefs.NotFound(errors.New("mismatched platform"))
+		}
+		if options.Platform != nil && desc.Platform != nil && !platform.Match(*options.Platform) {
+			return nil, errdefs.NotFound(errors.New("mismatched platform"))
+		}
+
```

Comment:
`!platform.Match(*options.Platform)` <- wrong argument

`platform` is a platform matcher that's supposed to match the options.Platform (`platform = cplatforms.Only(*options.Platform)`).

We should check if the descriptor's platform matches, so:
`!platform.Match(*desc.Platform)`

Reply:
```suggestion
		if options.Platform != nil {
		    if desc.Platform == nil || !platform.Match(*desc.Platform){
			    return nil, errdefs.NotFound(errors.New("mismatched platform"))
		    }
		}
```